File: cpp_tf2/src/static_broadcaster.cpp
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "tf2_ros/static_transform_broadcaster.h"
#include "geometry_msgs/msg/transform_stamped.hpp"

// #include "tf2/LinearMath/Quaternion.h" // usefull utility for Quaternion operations


class StaticBroadcasterNode : public rclcpp::Node {
public:
  explicit StaticBroadcasterNode()
  : Node("static_broadcaster_demo") {

    parent_frame_ = declare_parameter<std::string>("parent_frame", "");
    child_frame_ = declare_parameter<std::string>("child_frame", "");

    if (parent_frame_.empty()) {
      RCLCPP_ERROR_STREAM(
          this->get_logger(),
          "'parent_frame' parameter is required"
      );
    }

    if (child_frame_.empty()) {
      RCLCPP_ERROR_STREAM(
          this->get_logger(),
          "'child_frame' parameter is required"
      );
    }

    static_tf_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(this);
    geometry_msgs::msg::TransformStamped t_;

    this->make_transforms();
  }

private:

  void make_transforms() {
    t_.header.stamp = this->get_clock()->now();
    t_.header.frame_id = parent_frame_;
    t_.child_frame_id = child_frame_;

    t_.transform.translation.x = 0.5;
    t_.transform.translation.y = 0.0;
    t_.transform.translation.z = 1.0;

    t_.transform.rotation.x = 0.0;
    t_.transform.rotation.y = 0.0;
    t_.transform.rotation.z = 0.0;
    t_.transform.rotation.w = 1.0;

    static_tf_->sendTransform(t_);
  }

  std::string parent_frame_;
  std::string child_frame_;

  std::shared_ptr<tf2_ros::StaticTransformBroadcaster> static_tf_;
  geometry_msgs::msg::TransformStamped t_;
};


int main(int argc, char* argv[]) {

  rclcpp::init(argc, argv);

  auto node = std::make_shared<StaticBroadcasterNode>();

  rclcpp::spin(node);
  rclcpp::shutdown();

  return 0;
}

File: python_tf2/python_tf2/static_broadcaster.py
import sys

import rclpy
from rclpy.node import Node
from rcl_interfaces.msg import ParameterDescriptor

from tf2_ros.static_transform_broadcaster import StaticTransformBroadcaster
from geometry_msgs.msg import TransformStamped

class StaticBrodcaster(Node):

    def __init__(self):
        super().__init__("static_broadcaster_demo")

        self.__parent_frame = self.declare_parameter(
            "parent_frame",
            "",
            ParameterDescriptor(description="id of the parent frame.")
        ).get_parameter_value().string_value

        self.__child_frame = self.declare_parameter(
            "child_frame",
            "",
            ParameterDescriptor(description="id of the parent frame.")
        ).get_parameter_value().string_value

        if (self.__parent_frame == ""):
            self.get_logger().error("'parent_frame' parameter is required.")
            sys.exit(1)

        if (self.__child_frame == ""):
            self.get_logger().error("'child_frame' parameter is required.")
            sys.exit(1)

        self.__static_tf = StaticTransformBroadcaster(self)

        self.make_transforms()


    def make_transforms(self):
        self.get_logger().info(f"Starting static brodcaster from {self.__child_frame} -> {self.__parent_frame}")

        t = TransformStamped()

        t.header.stamp = self.get_clock().now().to_msg()
        t.header.frame_id = self.__parent_frame
        t.child_frame_id = self.__child_frame

        # Note: Keep in mind that this transform define
        # where is the `child` frame with respect to the `parent` frame

        t.transform.translation.x = 2.0
        t.transform.translation.y = 0.0
        t.transform.translation.z = 0.0

        t.transform.rotation.x = 0.0
        t.transform.rotation.y = 0.0
        t.transform.rotation.z = 0.0
        t.transform.rotation.w = 1.0

        self.__static_tf.sendTransform(t)


def main(args=None) -> None:

    rclpy.init(args=args)

    node = StaticBrodcaster()

    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.try_shutdown()


if __name__ == "__main__":
    main()
